tasks/bundle/index.js

```diff
@@ -0,0 +1,204 @@
+// TODO: add sourcemaps?
+// NOTE: BREAKING CHANGE!
+// dist/scripts/helix-ui.browser[.min].js -> dist/js/helix-ui[.min].js
+
+import { parallel, task } from 'gulp';
+import { rollup } from 'rollup';
+import { argv } from 'yargs';
+
+import { pkg, srcDir, distDir } from '../config';
+
+import { terserPlugin, es5plugins, es6plugins } from './plugins';
+
+const outDir = `${distDir}/js`;
+const flags = {
+    '--prod': 'generate additional minified bundles'
+};
+
+/**
+ * Generate CJS module bundle
+ *
+ * Syntax:       ES5
+ * Module:       CJS
+ * package.json: "main"
+ */
+async function bundleCJS () {
+    let bundle = await rollup({
+        input: `${srcDir}/_bundle.es5.js`,
+        plugins: es5plugins,
+    })
+
+    return bundle.write({
+        file: pkg.main,
+        format: 'cjs',
+    });
+}
+bundleCJS.description = 'Generate modules compatible with legacy bundlers';
+task('bundle:module:CJS', bundleCJS);
+
+/**
+ * Generate ESM module bundle
+ *
+ * Syntax:       ES5
+ * Module:       ESM
+ * package.json: "module"
+ */
+async function bundleESM () {
+    let bundle = await rollup({
+        input: `${srcDir}/_bundle.es5.js`,
+        plugins: es5plugins,
+    })
+
+    return bundle.write({
+        file: pkg.module,
+        format: 'esm',
+    });
+}
+bundleESM.description = 'Generate modules compatible with modern bundlers';
+task('bundle:module:ESM', bundleESM);
+
+/**
+ * Generate IE-compatible browser bundles
+ *
+ * Syntax:       ES5
+ * Module:       UMD
+ * package.json: "unpkg"
+ *
+ * @example
+ * <script [nomodule] src="..."></script>
+ */
+async function browserUMD () {
+    const inputOpts = {
+        input: `${srcDir}/_bundle.es5.js`,
+        plugins: es5plugins,
+    };
+
+    const outputOpts = {
+        file: pkg.unpkg,
+        format: 'umd',
+        name: 'HelixUI',
+        sourcemap: false,
+    };
+
+    let promiseMin = Promise.resolve();
```

Comment:
default value to a resolved promise (will be replaced with an unresolved promise if `--prod` flag used)

Reply:
TODO: add comment noting the same thing as the PR comment